components/org.wso2.carbon.identity.application.authenticator.apple/src/main/java/org/wso2/carbon/identity/application/authenticator/apple/AppleAuthenticator.java

```diff
@@ -145,8 +151,13 @@ protected void initiateAuthenticationRequest(HttpServletRequest request, HttpSer
                 String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
                 String authorizationEP = getAuthorizationServerEndpoint(authenticatorProperties);
                 String callbackUrl = getCallbackUrl(authenticatorProperties);
-                String state = getStateParameter(context, authenticatorProperties);
+                if (Boolean.parseBoolean((String) context.getProperty(IS_API_BASED))) {
```

Comment:
when this `IS_API_BASED` property is set in the context?

Reply:
This is being set at the framework[1].

[1] https://github.com/wso2/carbon-identity-framework/blob/3782fee47c7c612d37f955c12c5f70317c020e4b/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java#L791-L794

Reply:
So why can't we use the same constant defined in the framework?

Reply:
This concern was addressed with https://github.com/wso2-extensions/identity-outbound-auth-apple/pull/9/commits/9f2bc6ff1d342b6c4ceeec138af629a4634850a5